Dockerfile.jammy

```diff
@@ -0,0 +1,44 @@
+FROM docker-dev.yelpcorp.com/jammy_yelp:latest
+
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get --yes --force-yes install \
+      autoconf bison build-essential curl fakeroot gawk git-core libffi-dev \
+      libffi7 libgdbm-dev libncurses5-dev libpcre3-dev \
+      libreadline-dev libsqlite3-dev libxml2-dev libxslt1-dev \
+      pkg-config ruby screen sudo wget zlib1g zlib1g-dev && \
+    apt-get clean
+
+RUN rm -f /bin/sh && \
+    ln -s /bin/bash /bin/sh && \
+    touch "/etc/default/puppet" && \
+    mkdir -p /opt /package /etc/puppet /var/lib/puppet /var/cache/omnibus && \
+    echo 'gem: --no-document' > /etc/gemrc && \
+    echo 'install: --no-ri --no-rdoc' >> /etc/gemrc && \
+    echo 'update: --no-ri --no-rdoc' >> /etc/gemrc && \
+    cp /etc/gemrc /.gemrc && \
+    cp /etc/gemrc /root/.gemrc && \
+    cp /etc/gemrc /package/.gemrc
+
+ADD vendor/ruby-build-20211203.tar.gz /tmp
+
+# PEREL-2701: Download and install libssl-dev 1.1 from impish, since jammy
```

Comment:
I am not really a fan of this.  Do we have an idea of if/when this might be fixed upstream?  I'd rather wait a couple weeks than merge this now.  If we do merge this we probably want a ticket to clean this up that blocks finishing the jammy project

Reply:
...I guess if this is just to build the package it's not too bad, as long as we don't need to pull in the impish package to run this

Reply:
Yeah it's only as a build-time dependency (it uses `libssl1.1` at runtime), and I'm definitely not a fan either... I wish they published both `libssl1.1-dev` and `libssl3-dev` or something like they do for runtime packages ([libssl1.1](https://packages.ubuntu.com/jammy/libssl1.1) and [libssl3](https://packages.ubuntu.com/jammy/libssl3) do both exist alongside each other).

The problem is that I'm not sure how long this will take before it's fixed, and if it is fixed I expect it will only be for the latest ruby version (3.1.0 I suppose?). Since ruby releases come out on Christmas, I'm happy to wait until after this one and try out 3.1.0, but without https://github.com/ruby/openssl/issues/369 and something like https://github.com/ruby/ruby/pull/4904 merged in and released as a version I don't think it'll work yet. There's active work on this though, as https://github.com/ruby/openssl/pull/399#issuecomment-997965637 was added just yesterday (!)

Another option I was toying with was unvendoring ruby itself from this package and using the system ruby instead. I think that'd be nice, but seems like a big undertaking to do (and especially if doing so for all OS releases, not just jammy). I think that also somewhat defeats the original purpose of using an omnibus package here to include all the dependencies all together, although there are things like libssl/libxml/libxslt/virt-what that this depends on anyway, so maybe that's not that big of a difference. As far as I know, ruby doesn't have something like virtualenvs to make a more self-contained runtime with a specific set of gems, but maybe bundler into a separate directory would accomplish something similar.